opta/core/helm_cloud_client.py

```diff
@@ -16,39 +27,114 @@ def __init__(self, layer: "Layer"):
         super().__init__(layer)
 
     def get_remote_config(self) -> Optional["StructuredConfig"]:
-        return None
+        set_kube_config(self.layer)
+        load_opta_kube_config()
+        v1 = CoreV1Api()
+        secret_name = f"opta-config-{self.layer.state_storage()}"
+        secrets: V1SecretList = v1.list_namespaced_secret(
+            "default", field_selector=f"metadata.name={secret_name}"
+        )
+        if len(secrets.items) == 0:
+            return None
+        secret: V1Secret = secrets.items[0]
+        return json.loads(base64.b64decode(secret.data["config"]).decode("utf-8"))
 
     def upload_opta_config(self) -> None:
-        if "local" in REGISTRY[self.layer.cloud]["backend"]:
-            providers = self.layer.gen_providers(0)
-            local_path = providers["terraform"]["backend"]["local"]["path"]
-            real_path = os.path.dirname(os.path.realpath(local_path))
-            logger.warning(
-                f"The terraform state is stored locally, make sure to keep the files in {real_path}"
-            )
+        set_kube_config(self.layer)
+        load_opta_kube_config()
+        v1 = CoreV1Api()
+        secret_name = f"opta-config-{self.layer.state_storage()}"
+        create_secret_if_not_exists("default", secret_name)
+        current_secret_object: V1Secret = v1.read_namespaced_secret(
+            secret_name, "default"
+        )
+        current_secret_object.data = current_secret_object.data or {}
+        current_secret_object.data["config"] = base64.b64encode(
+            json.dumps(self.layer.structured_config()).encode("utf-8")
+        ).decode("utf-8")
+        v1.replace_namespaced_secret(secret_name, "default", current_secret_object)
+
         return None
 
     def delete_opta_config(self) -> None:
-        return None
+        set_kube_config(self.layer)
+        load_opta_kube_config()
+        v1 = CoreV1Api()
+        secret_name = f"opta-config-{self.layer.state_storage()}"
+        if check_if_secret_exists("default", secret_name):
+            v1.delete_namespaced_secret(secret_name, "default")
 
     def delete_remote_state(self) -> None:
-        return None
+        v1 = CoreV1Api()
+        secret_name = f"tfstate-default-{self.layer.state_storage()}"
```

Comment:
no need to have `default` here since it's a namespace scoped resource

Reply:
That's how terraform names the secret-- we got no choice :(